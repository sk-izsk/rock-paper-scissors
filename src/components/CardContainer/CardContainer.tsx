import clsx from 'clsx';
import React from 'react';
import { createUseStyles } from 'react-jss';
import { Card, CardContent } from 'ui-neumorphism';
import { CustomTheme } from '../../theme';

export interface CardContainerProps {
  inset?: boolean;
  cardAction?: JSX.Element;
  cardContentStyle?: string;
  cardStyle?: string;
  cardLoading?: boolean;
  cardBordered?: boolean;
}
const useStyles = createUseStyles((theme: CustomTheme) => ({
  card: {
    padding: theme.spacing(1),
  },
}));

const CardContainer: React.FC<CardContainerProps> = ({
  inset,
  children,
  cardAction,
  cardContentStyle,
  cardStyle,
  cardLoading,
  cardBordered,
}) => {
  const classes = useStyles();
  return (
    <Card
      className={clsx([classes.card, cardStyle && cardStyle])}
      loading={cardLoading && cardLoading}
      bordered={cardBordered && cardBordered}
      inset={inset}
    >
      <CardContent className={cardContentStyle ? cardContentStyle : undefined}>{children}</CardContent>
      {cardAction && cardAction}
    </Card>
  );
};

export { CardContainer };
