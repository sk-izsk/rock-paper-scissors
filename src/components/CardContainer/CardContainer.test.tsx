import { render } from '@testing-library/react';
import React from 'react';
import { CardContainer } from './CardContainer';

const mockData = {
  inset: true,
  cardAction: <div>mock</div>,
  cardLoading: true,
  cardBordered: true,
};

test('CardContainer renders correctly', () => {
  const { asFragment } = render(<CardContainer {...mockData} />);
  expect(asFragment()).toMatchSnapshot();
});
