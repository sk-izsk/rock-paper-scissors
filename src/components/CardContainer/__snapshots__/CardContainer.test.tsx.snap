// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="_28vIk _2bGnb _2Xmdp _1SFBz _12WTq makeStyles-card-1"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="0"
      class="_2M5iY _20rTC "
      role="progressbar"
      style="height: 4px; border-radius: 8px;"
    >
      <div
        class="_1wWaA _1PN0w"
        style="border-radius: 8px;"
      >
        <div
          class="_29wbC _2wIvQ"
          id="_07040_"
          style="width: 100%; height: 4px; border-radius: 8px;"
        />
      </div>
    </div>
    <div
      class="_2So3v _2uw1B "
    />
    <div>
      mock
    </div>
  </div>
</DocumentFragment>
`;
